test/e2e/preset_test.go

```diff
@@ -370,6 +398,52 @@ func validateInferenceResource(workspaceObj *kaitov1alpha1.Workspace, expectedRe
 	})
 }
 
+// Logic to validate he annotations of the workspace and deployment, as well as the corresponding controller revision
+func validateRevision(workspaceObj *kaitov1alpha1.Workspace, revisionStr string) {
+	By("Checking the revisions of the resources", func() {
+		Eventually(func() bool {
+			dep := &appsv1.Deployment{
+				ObjectMeta: metav1.ObjectMeta{
```

Comment:
I don't think you need to add ObjectMeta for the variable that will be used in the Get call. The same for line 421.